src/wp-includes/class-wp-query.php

```diff
@@ -814,18 +814,36 @@ public function parse_query( $query = '' ) {
 			$qv['p'] = (int) $qv['p'];
 		}
 
-		$qv['page_id']  = is_scalar( $qv['page_id'] ) ? absint( $qv['page_id'] ) : 0;
-		$qv['year']     = is_scalar( $qv['year'] ) ? absint( $qv['year'] ) : 0;
-		$qv['monthnum'] = is_scalar( $qv['monthnum'] ) ? absint( $qv['monthnum'] ) : 0;
-		$qv['day']      = is_scalar( $qv['day'] ) ? absint( $qv['day'] ) : 0;
-		$qv['w']        = is_scalar( $qv['w'] ) ? absint( $qv['w'] ) : 0;
-		$qv['m']        = is_scalar( $qv['m'] ) ? preg_replace( '|[^0-9]|', '', $qv['m'] ) : '';
-		$qv['paged']    = is_scalar( $qv['paged'] ) ? absint( $qv['paged'] ) : 0;
-		$qv['cat']      = preg_replace( '|[^0-9,-]|', '', $qv['cat'] ); // Array or comma-separated list of positive or negative integers.
-		$qv['author']   = is_scalar( $qv['author'] ) ? preg_replace( '|[^0-9,-]|', '', $qv['author'] ) : ''; // Comma-separated list of positive or negative integers.
-		$qv['pagename'] = is_scalar( $qv['pagename'] ) ? trim( $qv['pagename'] ) : '';
-		$qv['name']     = is_scalar( $qv['name'] ) ? trim( $qv['name'] ) : '';
-		$qv['title']    = is_scalar( $qv['title'] ) ? trim( $qv['title'] ) : '';
+		$qv['page_id']             = is_scalar( $qv['page_id'] ) ? absint( $qv['page_id'] ) : 0;
+		$qv['year']                = is_scalar( $qv['year'] ) ? absint( $qv['year'] ) : 0;
+		$qv['monthnum']            = is_scalar( $qv['monthnum'] ) ? absint( $qv['monthnum'] ) : 0;
+		$qv['day']                 = is_scalar( $qv['day'] ) ? absint( $qv['day'] ) : 0;
+		$qv['w']                   = is_scalar( $qv['w'] ) ? absint( $qv['w'] ) : 0;
+		$qv['m']                   = is_scalar( $qv['m'] ) ? preg_replace( '|[^0-9]|', '', $qv['m'] ) : '';
+		$qv['paged']               = is_scalar( $qv['paged'] ) ? absint( $qv['paged'] ) : 0;
+		$qv['cat']                 = preg_replace( '|[^0-9,-]|', '', $qv['cat'] ); // Array or comma-separated list of positive or negative integers.
+		$qv['author']              = is_scalar( $qv['author'] ) ? preg_replace( '|[^0-9,-]|', '', $qv['author'] ) : ''; // Comma-separated list of positive or negative integers.
+		$qv['pagename']            = is_scalar( $qv['pagename'] ) ? trim( $qv['pagename'] ) : '';
+		$qv['name']                = is_scalar( $qv['name'] ) ? trim( $qv['name'] ) : '';
+		$qv['title']               = is_scalar( $qv['title'] ) ? trim( $qv['title'] ) : '';
+		$qv['feed']                = is_scalar( $qv['feed'] ) ? trim( $qv['feed'] ) : '';
+		$qv['attachment']          = is_scalar( $qv['attachment'] ) ? $qv['attachment'] : '';
+		$qv['author_name']         = is_scalar( $qv['author_name'] ) ? $qv['author_name'] : '';
+		$qv['author__in']          = is_array( $qv['author__in'] ) ? $qv['author__in'] : array();
```

Comment:
I think this can be discarded, but I wonder if it'd be worthwhile using `wp_parse_list()` for these to coerce `author__in=author1` (and the other QVs) into an array. `wp_parse_id_list()` could also be used for `post__in` to also `absint()` them early.

```suggestion
		$qv['author__in']          = wp_parse_list( $qv['author__in'] );
```